### PR TITLE
Bw 526

### PR DESCRIPTION
**Pull request by milansar**

### CODE REVIEW

**Comment by milansar** [comment]
@anihalaney , Pull request is ready for your review

File: functions/services/achievement.service.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            return Utils.throwError(error);
+        }
+    }
+
+
+    /**
+     * getAchievementById
+     * return Achievement
+     */
+    static async getAchievementById(id: string): Promise<any> {
```
Comment (milansar) [inline functions/services/achievement.service.ts]: rename to getAchievementByUserId

File: firestore-rules.txt [hunk 2]
```diff
@@ -131,6 +131,12 @@ service cloud.firestore {
       // document can only be modified by API , triggers
         allow read,get:  if isUserAuthenticated(userId);          
     }
+
+    match /achievements {
```
Comment (milansar) [inline firestore-rules.txt]: When we are reading achievements through API from apps, we do not need achievements rights on store

File: projects/shared-library/src/lib/core/services/achievement.service.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+@Injectable()
+export class AchievementService {
+  constructor(
+    private http: HttpClient
+    ) {
+  }
+
+  getAchievements(): Observable<AchievementRule[]> {
+    const url: string = CONFIG.functionsUrl + '/app/achievement';
+    return this.http.get<AchievementRule[]>(url);
```
Comment (milansar) [inline projects/shared-library/src/lib/core/services/achievement.service.ts]: Should model be a AchievementRule or Acheivement?

File: projects/shared-library/src/lib/core/services/achievement.service.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+@Injectable()
+export class AchievementService {
+  constructor(
+    private http: HttpClient
+    ) {
+  }
+
+  getAchievements(): Observable<AchievementRule[]> {
+    const url: string = CONFIG.functionsUrl + '/app/achievement';
```
Comment (milansar) [inline projects/shared-library/src/lib/core/services/achievement.service.ts]: Use template string

File: projects/shared-library/src/lib/core/services/achievement.service.ts (hunk 3 above)
Comment (yugank1991) [reply projects/shared-library/src/lib/core/services/achievement.service.ts]: AchievementRule

File: functions/utils/push-notifications.ts [hunk 5]
```diff
@@ -120,9 +120,17 @@ export class PushNotification {
                     msg_data = { 'messageType': pushNotificationRouteConstants.QUESTION_NOTIFICATIONS };
                     result = await PushNotification
                         .sendNotificationToDevices(currentTurnPlayerId, 'Question Status Update',
-                        data, msg_data);
+                            data, msg_data);
                     console.log('result', result);
                     break;
+                case pushNotificationRouteConstants.ACHIEVEMENT_NOTIFICATION:
```
Comment (anihalaney) [inline functions/utils/push-notifications.ts]: we don't need push notifications for achievements

**Pull request merged**